File: 2023/string-230417_0512/1969.py
N, M = list(map(int, input().split()))

C = { 'A': 0, 'C': 0, 'G': 0, 'T': 0 }
cnt = [ C.copy() for _ in range(M) ]
for _ in range(N):
    for i, c in enumerate(input()):
        cnt[i][c] += 1

res = ""
dist = 0
for i in range(M):
    max_c = max(cnt[i], key=cnt[i].get)
    res += max_c
    for c in cnt[i]:
        if c != max_c:
            dist += cnt[i][c]
print(res)
print(dist) 

File: 2023/string-230417_0512/1969.cpp
#include <cstdio>

char s[51], res[51];
int cnt[50][128], dist;

const char C[] = { 'A', 'C', 'G', 'T' };

int main()
{
    int N, M;
    scanf("%d %d", &N, &M);
    for (int _ = 0; _ < N; _++) {
        scanf("%s", s);
        for (int i = 0; i < M; i++) {
            char c = s[i];
            cnt[i][c]++;
        }
    }

    for (int i = 0; i < M; i++) {
        char max_c = 'A';
        for (char c : C) {
            if (cnt[i][c] > cnt[i][max_c])  max_c = c;
        }
        res[i] = max_c;
        for (char c : C) {
            if (c != max_c)
                dist += cnt[i][c];
        }
    }
    
    printf("%s\n%d", res, dist);
}